pubsub/gcloud/aio/pubsub/__init__.py

```diff
@@ -4,7 +4,8 @@
 from gcloud.aio.pubsub.publisher_client import PublisherClient
 from gcloud.aio.pubsub.subscriber_client import SubscriberClient
 from gcloud.aio.pubsub.utils import PubsubMessage
+from gcloud.aio.pubsub.subscriber_message import Message
 
 
-__all__ = ['__version__', 'PublisherClient', 'PubsubMessage',
+__all__ = ['__version__', 'PublisherClient', 'Message', 'PubsubMessage',
```

Comment:
I wonder if we could reuse/augment the existing `PubsubMessage`? Or maybe that's getting ahead of ourselves at this point?

Perhaps `SubscriberMessage` might be more clear for now?

Reply:
Note to self: check if this is semantically the same message class as per google cloud API.

Reply:
Okay, so our implementation may eventually make these the same type.

However, a cursory look suggests that Google Cloud Pubsub _started off_ with publish-side and subscribe-side messages being the same type [from the protocol buffers], and eventually differentiated into different messages.

From the Publisher Client, the PubsubMessage protobuf wraps a binary blob with all the metadata needed to route it via gapic+gRPC -- https://github.com/googleapis/python-pubsub/blob/master/google/cloud/pubsub_v1/publisher/client.py#L357

From the Subscriber Client, the Subscriber.Message python object represents a parsed message, and makes it ready for user code to consume natively (blob -> attributes, data, meta) -- https://github.com/googleapis/python-pubsub/blob/master/google/cloud/pubsub_v1/subscriber/message.py#L53

Okay -- so, having said all of that -- _can we make both messages the same class?_ -- Maybe. But the similarity between the two may be an illusion. Really, they're symmetrical but have opposite orientations as the directionality converting either {python representation [or binary blob]} to|from {form compatible w gapic+gRPC|aiohttp+REST} matters.

Hmm ... however, having a symmetrical message class could also mean we would be able to use all the utilities to construct+parse messages into+outof either Python or API form ... which I suppose could serve some _retrying_ usecases [eg publish failure? np, poke inside message object, proofread+edit, publish again] and is _more functionality than is supported by native Google Cloud API_ ...

(However going from subscriber message blob -> blob for resubmission back into a topic would ofc be an anti-pattern heeh.)

Okay -- answer _for this PR_ -- no, let's not do that right now.

In future: I think _we literally have a deadlettering usecase_ that _might benefit_ from a symmetrical message class (I say "might" because we're nacking subscriber messages due to a downstream failure -- rather than some failure within the current app, so we're actually on the wrong side of the retry here). Let's take a look at how much more we can do with the existing deadlettering facilities, as well as bearing symmetrical messages in mind, as we work through our stack (deferred decision).